sympy/core/numbers.py

```diff
@@ -2015,7 +2015,7 @@ def __new__(cls, i):
             ival = int(i)
         except TypeError:
             raise TypeError(
-                'Integer can only work with integer expressions.')
+                'Integer can only work with expressions that can be rounded off to integer type.')
```

Comment:
Perhaps a shorter message like ("Argument %s of Integer should be of numeric type." % expr) would suffice.

Reply:
@jksuom In your suggested message, if we input `Integer(oo)`, then the error message will be : `Argument oo of Integer should be of numeric type`. Don't you think it is a bit absurd that the message is saying -- "oo should be of numeric type"?
Something like `Argument of Integer should be of numeric type` will be okay, what do you think?

Reply:
It may be useful to see the actual argument though perhaps not necessary. I think it is ok to drop the %s bit. Or it could be at the end: "... type, got %s".